gamification-evm-services/src/main/java/io/meeds/evm/gamification/scheduling/task/ERC20TransferTask.java

```diff
@@ -37,108 +40,158 @@
+          if (StringUtils.isBlank(rule.getEvent().getProperties().get(Utils.LAST_ID_PROCCED))) {
+            if (evmTransactionService.getTransactionByContractAddressAndNetworkIdOrderByIdDesc(contractAddress, networkId) != null) {
+              lastIdProcced = evmTransactionService.getTransactionByContractAddressAndNetworkIdOrderByIdDesc(contractAddress, networkId).getId();
+            }
+            broadcastEvmActionEvent(lastIdProcced.toString(), rule.getId().toString());
+          } else {
+            lastIdProcced = Long.parseLong(rule.getEvent().getProperties().get(Utils.LAST_ID_PROCCED));
+          }
+          List<EvmTransaction> transactions = evmTransactionService.getTransactionsByContractAddressAndNetworkIdFromId(contractAddress, networkId, lastIdProcced);
+          if (CollectionUtils.isNotEmpty(transactions)) {
+            transactions.forEach(transaction -> {
+              try {
+                if (trigger.equals(Utils.SEND_TOKEN_EVENT) || trigger.equals(Utils.RECEIVE_TOKEN_EVENT)
+                    || (trigger.equals(Utils.HOLD_TOKEN_EVENT)
+                        && isValidHoldingToken(transaction,
+                                               Long.parseLong(rule.getEvent().getProperties().get(Utils.DURATION)),
+                                               contractAddress,
+                                               blockchainNetwork))) {
+                  EvmTrigger evmTrigger = new EvmTrigger();
+                  evmTrigger.setTrigger(trigger);
+                  evmTrigger.setType(Utils.CONNECTOR_NAME);
+                  evmTrigger.setTransactionHash(transaction.getTransactionHash());
+                  evmTrigger.setTransactionId(transaction.getId());
+                  evmTrigger.setRuleId(rule.getId());
+                  evmTrigger.setContractAddress(contractAddress);
+                  evmTrigger.setBlockchainNetwork(blockchainNetwork);
+                  evmTrigger.setAmount(transaction.getAmount());
+                  evmTrigger.setNetworkId(networkId.toString());
+                  evmTrigger.setSentDate(transaction.getSentDate());
+                  if (trigger.equals(Utils.SEND_TOKEN_EVENT)) {
+                    evmTrigger.setWalletAddress(transaction.getFromAddress());
+                    evmTrigger.setTargetAddress(transaction.getToAddress());
+                  } else {
+                    if (trigger.equals(Utils.RECEIVE_TOKEN_EVENT)) {
+                      evmTrigger.setTargetAddress(transaction.getFromAddress());
+                    }
+                    evmTrigger.setWalletAddress(transaction.getToAddress());
+                  }
+                  evmTriggerService.handleTriggerAsync(evmTrigger);
+                  broadcastEvmActionEvent(transaction.getId().toString(), rule.getId().toString());
+                }
+              } catch (Exception e) {
+                LOG.warn("Error broadcasting event", e);
```

Comment:
Please give more details, at least the transaction hash for troubleshooting
```suggestion
                LOG.warn("Error broadcasting EVM event for transaction {} and trigger {}", transaction.getTransactionHash(), trigger, e);
```